docs/CLOUD_RECORDING.md

```diff
@@ -0,0 +1,284 @@
+
+We will check that these inputs are not empty. If they are return a Bad Request response using our helper function from `utils/sendResponse.ts`. 
+
+Our endpoints will need the following information in the body:
+- `api/recording/start.json` will need `uid` and `channel`
+- `api/recording/query.json` will need `sid` and `resourceId`
+- `api/recording/stop.json` will need `uid`, `channel`, `sid`, and `resourceId`
+
+If everything executes properly, at the end we send a successful response using our helper function with data we will fill in later.
+
+```ts
+import type { APIContext } from "astro";
+
+export async function POST({ request }: APIContext) {
+    const { uid, channel } = await request.json()
+
+    if (!channel) {
+        return sendBadRequest("channel is required")
+    }
+    if (!uid || uid === '') {
+        return sendBadRequest("uid is required")
+    }
+
+    return sendSuccessfulResponse("<return data>")
+}
+```
+
+## Make Request Helper Functions
+The rest of this guide will be focused on the implementation of Cloud Recording which includes calling the Agora API. To simplify the code and have a uniform request structure, we will set up a helper function in `utils/makeRequest.ts`. 
+
+The inputs will need the `method` (we will use GET and POST), `url`, `body`, and the `credential`. 
+
+This credential will be generated by another helper function in `utils/generateCredential.ts`. This is a base-64 encoded credential using the Customer ID and Customer Secret.
+
+```ts
+export const generateCredential = () => {
+  const credential = import.meta.env.CUSTOMER_ID + ":" + import.meta.env.CUSTOMER_SECRET
+
+  const base64_credential = btoa(credential)
+  return base64_credential
+}
+```
+
+The `makeRequest` function defines all the headers, executes the request, and returns the response. 
+
+```ts
+export const makeRequest = async (method: string, url: string, credential: string, body?: string) => {
+    const headers = new Headers({
+        "Authorization": "basic " + credential,
+        "Content-Type": "application/json",
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": `${method}, OPTIONS`,
+    });
+
+    const res = await fetch(url, {
+        method: method,
+        headers: headers,
+        body: body
+    })
+    if (!res.ok) {
+        console.log(await res.text())
+        throw new Error("Failed to make request")
+    }
+    return res
+}
+```
+
+## Generate Resource
+The first step in the cloud recording process is to generate a resource for cloud recording. Behind the scenes this spins up a backend Agora service that has the capability of recording your video calls. 
+
+> Your cloud recording needs to be started within five minutes, or else the resource will shut down.
```

Comment:
This note is fairly important for developers to know but feels like it gets lost between the paragraphs.

Reply:
I couldn't really find a better place to put it. I figured making it a blockquote will make it stand out more. But added the word "Warning" to make it even more clear

> Warning: Your cloud recording needs to be started within five minutes, or else the resource will shut down.